bootstrap/kubeadm/api/v1beta1/kubeadmconfig_webhook.go

```diff
@@ -63,7 +63,7 @@ func (c *KubeadmConfig) ValidateDelete() error {
 }
 
 func (c *KubeadmConfigSpec) validate(name string) error {
-	allErrs := c.Validate()
+	allErrs := c.Validate(field.NewPath("spec"))
```

Comment:
Is passing the spec field inside for code or is it solving an actual problem? While going through the changes, the older version with e.g. `"spec" "x" "y" "z"` is a bit more clear than the current proposed changes

Reply:
This solves the problem that we are reporting wrong field paths in our field errors in some webhooks.

We have different places where KubeadmConfigSpec are used, e.g. in KCP. But today all our webhooks report that the error is in `.spec...` without considering that it's actually e.g. `.spec.kubeadmConfigSpec...`

Furthermore (in my opinion) it's a best practice to build up those field paths incrementally, so each function only has to append new relative paths and we don't have to look up and down complex call hierarchies to figure out if the resulting field error has a correct field path. We are e.g. using that in our ClusterClass validation quite extensively (and in my opinion successfully :) ).